crates/sui-storage/src/object_store/downloader/gcs.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::object_store::downloader::{get, Downloader, DEFAULT_USER_AGENT};
+use anyhow::Result;
+use async_trait::async_trait;
+use bytes::Bytes;
+use object_store::path::Path;
+use object_store::GetResult;
+use percent_encoding::{percent_encode, utf8_percent_encode, NON_ALPHANUMERIC};
+use reqwest::Client;
+use reqwest::ClientBuilder;
+use std::sync::Arc;
+
+#[derive(Debug)]
+struct GoogleCloudStorageClient {
+    client: Client,
+    bucket_name_encoded: String,
+}
+
+impl GoogleCloudStorageClient {
+    pub fn new(bucket: &str) -> Result<Self> {
+        let mut builder = ClientBuilder::new();
+        builder = builder.user_agent(DEFAULT_USER_AGENT);
+        let client = builder.https_only(false).build()?;
+        let bucket_name_encoded = percent_encode(bucket.as_bytes(), NON_ALPHANUMERIC).to_string();
+
+        Ok(Self {
+            client,
+            bucket_name_encoded,
+        })
+    }
+
+    async fn get(&self, path: &Path) -> Result<GetResult> {
```

Comment:
assuming we enable it, both gcs and s3 http endpoints support listing the contents of the bucket, so could add support for that later. (it's not supported in cloudflare r2)

Reply:
yeah definitely, we should be able to add list support as well. I looked at it, and seemed a little complicated to left it as a todo for later